packages/acceptance-tests/pkg-tests-specs/sources/node-modules.test.ts

```diff
@@ -1780,6 +1780,72 @@ describe(`Node_Modules`, () => {
     ),
   );
 
+  it(`should set the correct permissions on binaries in installed packages`,
+    makeTemporaryEnv(
+      {
+        dependencies: {
+          [`has-bin-entries`]: `1.0.0`,
+        },
+      },
+      {
+        nodeLinker: `node-modules`,
+      },
+      async ({path, run}) => {
+        await run(`install`);
+        const {mode} = await xfs.lstatPromise(npath.toPortablePath(`${path}/node_modules/has-bin-entries/bin.js`));
+        const permissions = (mode & 0o777).toString(8);
+        expect(permissions).toBe(process.platform === `win32` ? `666` : `755`);
+      },
+    ),
+  );
+
+  it(`should set the correct permissions on binaries on packages that were upgraded`,
+    makeTemporaryEnv(
+      {
+        dependencies: {
+          [`has-bin-entries`]: `1.0.0`,
+        },
+      },
+      {
+        nodeLinker: `node-modules`,
+      },
+      async ({path, run}) => {
+        await run(`install`);
+        await xfs.writeJsonPromise(`${path}/package.json` as PortablePath, {
+          dependencies: {
+            'has-bin-entries': `2.0.0`,
+          },
+        });
+        await run(`install`);
+
+        const {mode} = await xfs.lstatPromise(npath.toPortablePath(`${path}/node_modules/has-bin-entries/bin.js`));
+        const permissions = (mode & 0o777).toString(8);
+        expect(permissions).toBe(process.platform === `win32` ? `666` : `755`);
```

Comment:
Unsure of whether we should test for `win32` here since it's tying the tests with `cmd-shim`. Should I explicitly test for everything except `win32` perhaps?